src/fetchtastic/download/async_core.py

```
@@ -0,0 +1,486 @@
+
+        Parameters:
+            callback (CoreProgressCallback): Callable receiving (downloaded, total, filename). May be synchronous or return a coroutine.
+            downloaded (int): Number of bytes downloaded so far.
+            total (Optional[int]): Total number of bytes if known, otherwise None.
+            filename (str): Target filename being downloaded.
+        """
+        try:
+            result = callback(downloaded, total, filename)
+            if asyncio.iscoroutine(result):
+                await result
+        except Exception as e:
```

Ruff (0.15.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)